lib/reader.js

```diff
@@ -1,115 +1,53 @@
 'use strict';
 
-const JSONStream = require('JSONStream');
 const pack = require('browser-pack');
-const mergeStream = require('merge-stream');
-const stream = require('readable-stream');
-
-// Ensure output order is the same as input order. The sorter only sorts the
-// modules, not execution order. `order` is consumed by browserify's
-// `browser-pack`. The feeds already have an order, but they are always `0` as
-// each feed only have a single entrypoint.
-class OrderDecorator extends stream.Transform {
-    constructor (order) {
-        super({
-            objectMode: true,
-        });
-
-        this.order = order;
-    }
-
-    _transform (row, encoding, next) {
-        if (row.entry === true) {
-            row.order = this.order;
-        }
-        next(null, row);
-    }
+const getStream = require('get-stream');
+const assert = require('assert');
+
+function getRows (feeds, existMap = new Map()) {
+    return feeds
+        .reduce((acc, feed, order) => {
+            feed.forEach((row) => {
+                // add browserify.order and flatten
+                delete row.order;
+                if (row.entry === true) {
+                    row.order = order;
+                }
+                acc.push(row);
+            });
+            return acc;
+        }, [])
+        .filter(row => {
+            assert(row && row.id, `Expected feed entry to have an id, got "${row.id}"`);
+            if (existMap.has(row.id)) {
+                return false;
+            }
+            existMap.set(row.id, true);
+            return true;
+        })
+        .sort((a, b) => a.id.localeCompare(b.id));
 }
 
-class SortAndDedupe extends stream.Transform {
-    constructor () {
-        super({
-            objectMode: true,
-        });
-
-        this.rows = new Map();
-    }
-
-    _transform (row, encoding, next) {
-        // FIXME: This is too simplistic. We have to handle the case of rows
-        // having keys like `expose` set, and ensure the correct configuration
-        // bubbles up
-        if (row && row.id && !this.rows.has(row.id)) {
-            this.rows.set(row.id, row);
-        }
-        next();
-    }
 
-    _flush (next) {
-        Array
-            .from(this.rows.values())
-            .sort(compareId)
-            .forEach(row => this.push(row));
-        next();
-    }
-}
+module.exports = async (feeds, existMap) => {
```

Comment:
should talk about it at work, but we should plug in some way of doing logging

Reply:
How do you see logging done?

Reply:
I'm not sure... Either we should pass in an object with `info`, `debug` etc, ot we should return an event emitter.

@digitalsadhu and I started with a passing in an object containing the different functions yesterday, but it felt really clunky